sourmash/sourmash_args.py

```diff
@@ -63,17 +64,22 @@ def calculate_moltype(args, default=None):
     return moltype
 
 
-def load_query_signature(filename, ksize, select_moltype):
+def load_query_signature(filename, ksize, select_moltype, select_md5=None):
     try:
-        sl = signature.load_signatures(filename,
-                                       ksize=ksize,
-                                       select_moltype=select_moltype,
-                                       do_raise=True)
+        sl = load_file_as_signatures(filename, ksize=ksize,
+                                     select_moltype=select_moltype)
         sl = list(sl)
-    except (IOError, ValueError):
+    except (OSError, ValueError):
         error("Cannot open file '{}'", filename)
         sys.exit(-1)
 
+    if len(sl) and select_md5:
+        for sig in sl:
+            sig_md5 = sig.md5sum()
+            if sig_md5.startswith(select_md5.lower()):
+                sl = [sig]
+                break
```

Comment:
for large collections (databases?), should we check if there is more than one sig starting with the same MD5?

Reply:
yes, thanks for catching this!

I'm not sure what to do about this... for really large collections it could be quite slow. But then again, for large collections it's not clear you want to use this! So, yeah. I'll change the behavior and add a test.

Reply:
done in 11e519a8

Reply:
> I'm not sure what to do about this... for really large collections it could be quite slow. But then again, for large collections it's not clear you want to use this! So, yeah. I'll change the behavior and add a test.

Yeah... at this point there is no access to the databases anymore, where this info can be checked while loading or it's less expensive (at least for SBTs? I think it is for LCA too).